src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -3376,12 +3376,15 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
             return (ASR::asr_t *)ASR::down_cast<ASR::expr_t>(ASR::make_Cast_t(
             al, loc, arg, ASR::cast_kindType::LogicalToInteger,
             to_type, value));
-        } else if (!ASRUtils::is_integer(*type)) {
+        } else if (ASRUtils::is_integer(*type)) {
+            return (ASR::asr_t *)arg;
```

Comment:
So the kind will be retained here, right? If input is 32 bit integer then answer will also be 32 bit integer and if the input is 64 bit integer then answer will also be 64 bit integer.

Reply:
The kind would be retained for `int()`, whereas `float()` now gives only 64-bit numbers.

Reply:
Why the difference in the behaviour of `int()` and `float()`? Is it not good to keep the behaviour for `int()` and `float()` similar?